core/src/test/java/org/apache/iceberg/hadoop/TestCachingCatalog.java

```diff
@@ -252,9 +254,9 @@ public void testCacheExpirationEagerlyRemovesMetadataTables() throws IOException
     ticker.advance(HALF_OF_EXPIRATION);
     Assertions.assertThat(catalog.cache().asMap()).doesNotContainKey(tableIdent);
 
-    Arrays.stream(metadataTables(tableIdent)).forEach(metadataTable ->
-        Assert.assertFalse("When a data table expires, its metadata tables should expire regardless of age",
-            catalog.cache().asMap().containsKey(metadataTable)));
+    // Removal of metadata tables from cache is async, use awaitility
```

Comment:
Nit: This comment seems unnecessary, as we're using `awaitility` immediately below.